libnavigation-base/src/test/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObjectMapperTest.kt

```diff
@@ -0,0 +1,335 @@
+package com.mapbox.navigation.base.trip.model.roadobject
+
+import com.mapbox.geojson.Geometry
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.internal.factory.RoadObjectInstanceFactory
+import com.mapbox.navigation.base.trip.model.roadobject.border.CountryBorderCrossing
+import com.mapbox.navigation.base.trip.model.roadobject.border.CountryBorderCrossingAdminInfo
+import com.mapbox.navigation.base.trip.model.roadobject.border.CountryBorderCrossingInfo
+import com.mapbox.navigation.base.trip.model.roadobject.incident.Incident
+import com.mapbox.navigation.base.trip.model.roadobject.incident.IncidentCongestion
+import com.mapbox.navigation.base.trip.model.roadobject.incident.IncidentImpact
+import com.mapbox.navigation.base.trip.model.roadobject.incident.IncidentInfo
+import com.mapbox.navigation.base.trip.model.roadobject.incident.IncidentType
+import com.mapbox.navigation.base.trip.model.roadobject.restrictedarea.RestrictedArea
+import com.mapbox.navigation.base.trip.model.roadobject.reststop.RestStop
+import com.mapbox.navigation.base.trip.model.roadobject.reststop.RestStopType
+import com.mapbox.navigation.base.trip.model.roadobject.tollcollection.TollCollection
+import com.mapbox.navigation.base.trip.model.roadobject.tollcollection.TollCollectionType
+import com.mapbox.navigation.base.trip.model.roadobject.tunnel.Tunnel
+import com.mapbox.navigation.base.trip.model.roadobject.tunnel.TunnelInfo
+import com.mapbox.navigator.IncidentCongestionDescription
+import com.mapbox.navigator.MatchedRoadObjectLocation
+import com.mapbox.navigator.RoadObject
+import com.mapbox.navigator.RoadObjectMetadata
+import com.mapbox.navigator.RoadObjectProvider
+import com.mapbox.navigator.RouteAlertLocation
+import io.mockk.mockk
+import org.junit.Assert
+import org.junit.Test
+import java.util.Date
+
+private typealias SDKRouteAlertLocation =
+    com.mapbox.navigation.base.trip.model.roadobject.location.RouteAlertLocation
+
+private typealias SDKRoadObjectProvider =
+    com.mapbox.navigation.base.trip.model.roadobject.RoadObjectProvider
+
+class RoadObjectMapperTest {
+
+    private val shape: Geometry = Point.fromLngLat(LONGITUDE, LATITUDE)
+    private val location = SDKRouteAlertLocation(shape)
+
+    @Test
+    fun `tunnel entrance alert is parsed correctly`() {
+        val nativeObject = tunnel
+
+        val expected = Tunnel(
+            ID,
+            TunnelInfo("Ted Williams Tunnel"),
+            LENGTH,
+            location,
+            SDKRoadObjectProvider.MAPBOX,
+            nativeObject
+        )
+
+        val roadObject = RoadObjectInstanceFactory.buildRoadObject(nativeObject)
+
+        Assert.assertEquals(expected, roadObject)
+        Assert.assertEquals(expected.hashCode(), roadObject.hashCode())
+        Assert.assertEquals(expected.toString(), roadObject.toString())
+        Assert.assertEquals(RoadObjectType.TUNNEL, roadObject.objectType)
```

Comment:
```suggestion
        assertEquals(expected, roadObject)
        assertEquals(expected.hashCode(), roadObject.hashCode())
        assertEquals(expected.toString(), roadObject.toString())
        assertEquals(RoadObjectType.TUNNEL, roadObject.objectType)
```

Same thing applies to other tests below 👇

Reply:
done